spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -140,8 +140,19 @@ protected void doStart() {
 		if (isRunning()) {
 			return;
 		}
-		setRunning(true);
-		Object messageListener = getContainerProperties().getMessageListener();
+		ContainerProperties containerProperties = getContainerProperties();
+
+		if (!this.consumerFactory.isAutoCommit()) {
+			AckMode ackMode = containerProperties.getAckMode();
+			if (ackMode.equals(AckMode.BATCH) || ackMode.equals(AckMode.COUNT) || ackMode.equals(AckMode.COUNT_TIME)) {
+				Assert.state(containerProperties.getAckCount() > 0, "'ackCount' must be > 0");
```

Comment:
Well, I think I should study more about those `MANUAL`, but I don't understand your concern about `BATCH`.
Honestly it is used only from the same place as `COUNT` : 
```
if (ackMode.equals(AckMode.BATCH) || ackMode.equals(AckMode.COUNT) && countExceeded) {
	if (this.logger.isDebugEnabled()) {
		this.logger.debug("Committing in AckMode.COUNT because count " + this.count
					+ " exceeds configured limit of " + this.containerProperties.getAckCount());
	}
	commitIfNecessary();
	this.count = 0;
}
```
And, yes, `COUNT` isn't used anymore, too!

Don't we miss anything here?